### PR TITLE
optimization: use memcmp instead of strncmp for ASK, MOVED, etc.

### PR DESCRIPTION
**Pull request by zhuizhuhaomeng**

### CODE REVIEW

**Comment by zhuizhuhaomeng** [comment]
@bjosv any progress on this PR?

**Comment by bjosv** [comment]
Hi @zhuizhuhaomeng, thanks for the PR!
We have discussed it a bit a we like the `memcmp()` change that is better, but we think using `strlen()` is easier to read than `sizeof() - 1`.
Could we keep the `strlen()` parts for readability since the compiler seems to optimize it nicely anyway?
WDYT?

**Comment by zhuizhuhaomeng** [comment]
@bjosv I have updated the PR.

**Review by zuiderkwast — APPROVED** [review]
LGTM

**Review by bjosv — APPROVED** [review]
Thanks!

File: hircluster.c [hunk 1]
```diff
@@ -220,20 +220,20 @@ static int cluster_reply_error_type(redisReply *reply) {
 
     if (reply->type == REDIS_REPLY_ERROR) {
         if ((int)strlen(REDIS_ERROR_MOVED) < reply->len &&
-            strncmp(reply->str, REDIS_ERROR_MOVED, strlen(REDIS_ERROR_MOVED)) ==
-                0) {
+            memcmp(reply->str, REDIS_ERROR_MOVED,
+                   strlen(REDIS_ERROR_MOVED)) == 0) {
```
Comment (bjosv) [inline hircluster.c:224]: Fix clang format issue
```suggestion
            memcmp(reply->str, REDIS_ERROR_MOVED, strlen(REDIS_ERROR_MOVED)) ==
                0) {
```

**Pull request merged**